internal/api/handlers.go

```diff
@@ -39,6 +35,13 @@ func GetRecommendationSetList(c echo.Context) error {
 		}
 	}
 
+	if !is_user_authorized_for_resource(types.ResourceObject{
+		Cluster: c.QueryParam("cluster"),
+		Project: c.QueryParam("project"),
+	}, user_permissions) {
```

Comment:
How does this condition work when user is having cluster permissions but not having permissions for few projects?  
Does that mean user can get recommendations for projects when he is having permissions for cluster, correct?

Reply:
Yes, that's correct. If user have the cluster permission that means user is authorized for all the projects in that cluster.

Reply:
@patilsuraj767 In that case, do we need L40?

Reply:
Yes, we need it. Because it may happen user do not have any cluster permission but he has project level permission.